x/oracle/keeper/historic_price.go

```diff
@@ -296,6 +296,11 @@ func (k Keeper) AddHistoricPrice(
 	ak.updateAvgCounter(denom, exchangeRate, ctx.BlockTime())
 }
 
+func (k Keeper) GetHistoricAvgPrice(ctx sdk.Context, denom string) (sdk.Dec, error) {
+	ak := AvgKeeper{cdc: k.cdc, store: ctx.KVStore(k.storeKey), period: AvgPeriod, shift: AvgShift}
```

Comment:
let's have a Keeper function to construct the AvgKeeper, then we can simplify here;
```
return k.AvgKeeper(ctx).GetCurrentAvg(denom)
```